rpcs3/Emu/Cell/Modules/cellAvconfExt.cpp

```diff
@@ -73,21 +180,23 @@ s32 cellVideoOutGetGamma(u32 videoOut, vm::ptr<f32> gamma)
 	return CELL_OK;
 }
 
-s32 cellAudioInGetAvailableDeviceInfo(u32 count, vm::ptr<CellAudioInDeviceInfo> info)
+s32 cellAudioInGetAvailableDeviceInfo(u32 count, vm::ptr<CellAudioInDeviceInfo> device_info)
 {
-	cellAvconfExt.todo("cellAudioInGetAvailableDeviceInfo(count=0x%x, info=*0x%x)", count, info);
-	info->portType = CELL_AUDIO_IN_PORT_USB;
-	info->availableModeCount = 1;
-	info->state = CELL_AUDIO_IN_DEVICE_STATE_AVAILABLE;
-	info->deviceNumber = 0;
-	// Some games check if deviceId and type are the same.
-	info->deviceId = 0xDEADBEEF;
-	info->type = 0xBEEFDEAD;
-	info->availableModes[0].type = CELL_AUDIO_IN_CODING_TYPE_LPCM;
-	info->availableModes[0].channel = CELL_AUDIO_IN_CHNUM_NONE;
-	info->availableModes[0].fs = CELL_AUDIO_IN_FS_8KHZ | CELL_AUDIO_IN_FS_12KHZ | CELL_AUDIO_IN_FS_16KHZ | CELL_AUDIO_IN_FS_24KHZ | CELL_AUDIO_IN_FS_32KHZ | CELL_AUDIO_IN_FS_48KHZ;
-	strcpy(info->name, "USB Camera");
-	return 1; // number of available devices
+	cellAvconfExt.todo("cellAudioInGetAvailableDeviceInfo(count=0x%x, info=*0x%x)", count, device_info);
+
+	if (count > 16 || !device_info.addr())
+	{
+		return CELL_AUDIO_IN_ERROR_ILLEGAL_PARAMETER;
+	}
+
+	auto av_manager = fxm::get_always<avconf_manager>();
+
```

Comment:
Doesn't just changing count to be `count = std::min<u32>(count, av_manager->devices.size())` make things a bit easier? Simplifies the for loop comparison as well as the return statement.

Reply:
It does indeed, I went the overly complicated way.